flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMaster.java

```diff
@@ -571,9 +572,16 @@ private void releaseEmptyTaskManager(ResourceID resourceId) {
 	@Override
 	public CompletableFuture<RegistrationResponse> registerTaskManager(
 			final String taskManagerRpcAddress,
-			final TaskManagerLocation taskManagerLocation,
+			final TaskManagerUnresolvedLocation taskManagerUnresolvedLocation,
 			final Time timeout) {
 
+		final TaskManagerLocation taskManagerLocation;
+		try {
+			taskManagerLocation = TaskManagerLocation.fromUnresolvedLocation(taskManagerUnresolvedLocation);
```

Comment:
For data transmission, yes. But I think the TM's address is required to be resolvable for JM process anyway. Because we still need them for RPC communications.

Reply:
I guess it should be fine since the `TM` establishes the connection. But this is hairsplitting from my side. I'm good with keeping it as is.